src/context.c

```diff
@@ -140,6 +140,43 @@ static gchar* get_cmdline_bootname(void)
 	return bootname;
 }
 
+static gchar* get_custom_bootname(void)
+{
+	g_autoptr(GSubprocessLauncher) launcher = NULL;
+	g_autoptr(GSubprocess) handle = NULL;
+	g_autoptr(GDataInputStream) datainstream = NULL;
+	g_autoptr(GPtrArray) args_array = NULL;
+	g_autofree gchar *outline = NULL;
+	GInputStream *instream;
+
+	if (context->mock.custom_bootloader_output) {
```

Comment:
This prevents all the added code from being tested. Please replace this by actually testing the code.

Your config file already references the custom bootloader test script `bin/custom-bootloader-script`. This should be extended to actually return something on a `get-current` argument invocation.

All mock code should be removed.